### PR TITLE
Rework Molitz Beta to be more dynamic

### PR DESCRIPTION
**Pull request by maybeserendipitous**
[REWORK][INPUT WANTED]

## About the PR 

This is a rework of Molitz Beta's mechanics. The Resonance material property of Molitz Beta now plays a central role.
Increasing Resonance is dependent on Density and current Resonance.
Producing Oxygen Beta is dependent on Hardness, Radioactivity, and Neutron Radioactivity, and decreases Resonance in the process.

This will hopefully make Molitz Beta a viable, interesting alternative to pipeburns.
The power level of this rework needs some testing in a live environment, but there should be enough knobs to turn to balance it if necessary.

## Why's this needed? 

Molitz Beta is currently very linear. There is not much room for tradeoffs or experimentation, and it pays lip service to the Resonance property. I have never encountered an experienced Engineer who was excited to do a Molitz Beta burn, compared to a pipeburn, where Engineers are fairly willing. This rework aims to solve this.

Let's say you want to increase Resonance. Will you smelt ores together to increase density? Which material will you choose, and how will you get them? Will you try to get as close to the saturation threshold as possible to avoid loss, or would you rather use up the Resonance and blow it up again later?
Now let's say you want to increase Oxygen Beta output. Again, will you smelt ores together to increase Hardness? Will you avoid or embrace radioactivity? Maybe you'd even prefer farming Oxygen Beta instead of producing it mid-combustion, since higher temperatures increase the rate of Resonance loss.

All this and more must be considered when using Molitz Beta, and it's my hope that this will translate to more engaging gameplay.

## Changelog

```changelog
(u)maybeserendipitous
(*)Rework Molitz Beta to be more engaging
```

### CODE REVIEW

**Comment by Tarmunora** [comment]
Meh. Can't say I'm interested

**Comment by maybeserendipitous** [comment]
> Meh. Can't say I'm interested

Funny, that's how I feel about Molitz Beta currently. Twinsies!!

**Comment by Tarmunora** [comment]
After some discussion in admin chats, closing this.

**Review by Azrun — COMMENTED** [review]
What are the actual ranges of gas this ends up producing?   How many iterations of catalyzed atmos hotspot are to be expected? How does this compare to the previous implementation(s)? 


I don't agree that this implementation will meet the goals of this PR.  That said I wanted to give some feedback on how this was implemented in regards to content from possible future PRs.

File: code/modules/materials/Mat_MaterialProcs.dm [hunk 1]
```diff
@@ -389,7 +386,33 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
...
+			// max 15
+			var/g = BASE_SCALE + (t * (HIGH * (1 - hardness)) + (radioactive) + (MID * n_radioactive))
+
+			// scales amount of resonance lost
+			// max 17
+			var/loss = BASE_SCALE + (t * (MID * (1 - hardness)) + (MID * radioactive) + (HIGH * n_radioactive))
+
 			var/datum/gas/oxygen_agent_b/trace_gas = payload.get_or_add_trace_gas_by_type(/datum/gas/oxygen_agent_b)
 			payload.temperature = T0C // Greatly reduce temperature to simulate an endothermic reaction
 			// Itr: .18 Agent B, 20 oxy, 1.3 minutes per iteration, realisticly around 7-8 minutes per crystal.
```
[the remark this answers is not in the document]
Comment (maybeserendipitous) [reply code/modules/materials/Mat_MaterialProcs.dm:418]: I don't understand.

[the remark this answers is not in the document]
Comment (Azrun) [reply code/modules/materials/Mat_MaterialProcs.dm:418]: Comment outlines how much is released per iteration.  This is then used to determine how long an area would be catalyzed.  Theoretical best case.  Important when factoring in accessibility and availability of Molitz B.
